src/utils/session.ts

```diff
@@ -22,3 +22,26 @@ export function getPresenterAccountDetails(req: Request): Details {
 export function setPresenterAccountDetails(req: Request, details: Details) {
     req.session?.setExtraData(PRESENTER_ACCOUNT_SESSION_KEY, details);
 }
+
+export function generateUserDetails(req: Request): Details{
```

Comment:
I think this needs a more descriptive name. It doesn't generate the details, it populates them from the request.